### PR TITLE
fix: support other class component implement

### PR DESCRIPTION
**Pull request by agileago**
vue3 class component shape is here 

https://github.com/vuejs/vue-next/blob/7007ffb2c796d6d56b9c8e278c54dc1cefd7b58f/packages/runtime-core/src/component.ts#L122-L125

```typescript
export interface ClassComponent {
  new (...args: any[]): ComponentPublicInstance<any, any, any, any, any>
  __vccOpts: ComponentOptions
}
```

but the library `mount` type shape is not adjust it. so we should support other class component implement

### CODE REVIEW

**Comment by cexbrayat** [comment]
@agileago Thanks for the PR!

Would you mind adding a test to show what this fixes? You can check `mount.d.ts` for examples.

**Comment by agileago** [comment]
ok

File: test-dts/mount.d-test.ts [hunk 1]
```diff
@@ -211,6 +212,48 @@ class ClassComponent extends Vue {
...
+  changeCount(count: number) {
+    this.count.value = count
+  }
+
+  render() {
+    return h('div', `hello world ${this.count.value}`)
+  }
+}
+
+// @ts-expect-error changeMessage expects an argument
```
Comment (cexbrayat) [inline test-dts/mount.d-test.ts:252]: changeMessage -> changeCount

File: test-dts/mount.d-test.ts [hunk 2]
```diff
@@ -1,8 +1,9 @@
 import { expectError, expectType } from './index'
 import {
+  ComponentOptions,
   DefineComponent,
   defineComponent,
-  FunctionalComponent
+  FunctionalComponent, h, ref, SetupContext
```
Comment (cexbrayat) [inline test-dts/mount.d-test.ts:6]: nit: this should be formatted properly

File: test-dts/mount.d-test.ts [hunk 3]
```diff
@@ -211,6 +212,48 @@ class ClassComponent extends Vue {
...
+      props: this.defaultProps,
+      setup(props, ctx) {
+        const instance = new CompConstructor()
+        instance.props = props
+        instance.context = ctx
+        ctx.expose(instance)
+        return instance.render.bind(instance)
+      }
+    }
+  }
```
Comment (cexbrayat) [inline test-dts/mount.d-test.ts:232]: OOC: Is this something that developers really write? Why not extend the base `Vue` class?

Comment (agileago) [reply test-dts/mount.d-test.ts:235]: where is the Vue class? i use it in my library [vue3-oop ](https://github.com/agileago/vue3-oop)

Comment (cexbrayat) [reply test-dts/mount.d-test.ts:235]: `vue-class-component` exports a `Vue` class that components should extend (if I remember correctly).

**Comment by cexbrayat** [comment]
@agileago The type check tests are failing, so this change is probably breaking something for the type inference. Can you take a look?

**Comment by agileago** [comment]
git rebase is so hard...

**Pull request closed without merging**